### PR TITLE
Leaf/f5.2 hamburger menu v2

### LINKED ISSUE #138: F5.2 Hamburger Menu
As a general user of the site, I want to be able to navigate the site using a hamburger menu

Details: Add a hamburger menu to the header of the site. All button functionality should be included or prepared. 

Technical Details: 
- Front-end
- React
- MUI
- UI/UX

### PR DESCRIPTION
**Pull request by Leaversa**
### TODO ###
- Since routing has not been setup as of making this pr, hamburger menu items do not do anything #258.

### NOTES ###
- I made some deviations from the figma.
  - Underlines underneath menu items retain are the same color as text.
  - Underlines are also set as the full width of the side menu.
  - Account Settings has been shortened to Settings.
  - Text color has been set to text.secondary according to the MainTheme file.
  
Closes #138

### CODE REVIEW

File: meshapp/src/components/Header.tsx [hunk 1]
```diff
@@ -1,30 +1,13 @@
-import { AppBar, Grid, IconButton, Toolbar, Typography } from "@mui/material";
-
-import MenuIcon from "@mui/icons-material/TableRows";
-import NotificationsIcon from "@mui/icons-material/Notifications";
-
 import React from "react";
+import { AppBar, Grid, Toolbar, Typography } from "@mui/material";
+import SideMenu from "./SideMenu";
```
Comment (Darren52BT) [inline meshapp/src/components/Header.tsx:3]: This import does not appear to be working. Looks like you have a separate Components folder where this side menu is located.

**Review by Darren52BT — CHANGES_REQUESTED** [review]

File: meshapp/src/components/SideMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  Logout as LogoutIcon,
+} from "@mui/icons-material";
+
+type ListItemTypes = { text: String; Icon: React.ElementType };
+
+// TODO: Add routes to each menu item when routing is implemented
+const ListItemComponent = ({ text, Icon }: ListItemTypes) => (
+  <ListItem>
+    <ListItemButton
+      sx={{ borderBottom: "1px solid", borderColor: "text.secondary" }}
```
Comment (Darren52BT) [inline meshapp/src/components/SideMenu.tsx:30]: Though the figma does say text/secondary, it doesn't look like the dark theme part matches the figma, and the text is a bit hard to see. Do you think we should add another theme color to the palette to use specifically for the menu bar?

File: meshapp/src/components/SideMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  Receipt as ProfileIcon,
+  Inbox as MessagingIcon,
+  People as SwipingIcon,
+  Settings as SettingsIcon,
+  Logout as LogoutIcon,
+} from "@mui/icons-material";
+
+type ListItemTypes = { text: String; Icon: React.ElementType };
+
+// TODO: Add routes to each menu item when routing is implemented
```
Comment (misslame) [inline meshapp/src/components/SideMenu.tsx:26]: If youre going to add a ToDo, add a corresponding issue

File: meshapp/src/components/SideMenu.tsx (hunk 2 above)
Comment (Leaversa) [reply meshapp/src/components/SideMenu.tsx:30]: I left notes about the theme changes in my pr description. Additionally I think that palette changes aren't in the scope of the task and should be addressed in another issue.

**Review by misslame — APPROVED** [review]
LGTM

File: meshapp/src/components/SideMenu.tsx (hunk 3 above)
Comment (Leaversa) [reply meshapp/src/components/SideMenu.tsx:26]: I have it linked in my pr description

**Review by Darren52BT — APPROVED** [review]

**Pull request merged**